apps/sms/test/marionette/conversation_test.js

```diff
@@ -491,4 +503,83 @@ marionette('Conversation Panel Tests', function() {
       });
     });
   });
+
+  suite('Add recipients from thread', function() {
+    var conversationView;
+    setup(function() {
+      var thread = ThreadGenerator.generate({
+        numberOfMessages: 1
+      });
+      messagesApp.launch();
+
+      storage.setMessagesStorage([thread], ThreadGenerator.uniqueMessageId);
+      // Set empty contacts store.
+      storage.setContactsStorage();
+      var inboxView = new InboxView(client);
+      conversationView = inboxView.goToConversation(thread.id);
+      messagesApp.Composer.messageInput.sendKeys('message2');
+    });
+
+    test('Sms being written: Message and recipients should be kept',
+    function() {
+      conversationView.showOptions();
+      messagesApp.Menu.selectAppMenuOption('Add recipients');
+      // this also ensure we are on Composer view
+      messagesApp.getRecipient('+123');
+      client.waitFor(function() {
+        return messagesApp.Composer.messageInput.text() === 'message2';
+      });
+    });
```

Comment:
note that in your case it would be a good idea to return a NewMessageView (https://github.com/mozilla-b2g/gaia/blob/master/apps/sms/test/marionette/lib/views/new-message/view.js) from this new method ;)